### PR TITLE
[SWE-23] Forgot Password Frontend

### PR DESCRIPTION
**Pull request by sammyoto**
Still need to fix a few things. Overall logic is implemented, need some guidance on where to implement typescript into something like this @jackfales. Also, if you input an invalid email into the box and hit submit, react automatically tells you to input a valid email? Inputing "samgfales" for example will give a popup underneath the submit box. @jackfales @datderek Do you guys know where this comes from?

### CODE REVIEW

**Comment by datderek** [comment]
@sammyoto If you're talking about this tooltip, then that should just be a native HTML feature. Giving the input `type="email"` will enable that feature.
image

File: pages/forgot-password.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import Head from 'next/head';
+import React, { useState } from 'react';
+import styles from '../styles/Home.module.css';
+
+function ForgotPassword () {
+  const [message, setMessage] = useState('');
```
Comment (datderek) [inline pages/forgot-password.tsx:6]:
```suggestion
  const [errorMessage, setErrorMessage] = useState('');
```

File: pages/forgot-password.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    let email = formData.get("email")
+
+    // TODO implement email format checking
+    if (email.length === 0) {
+        setMessage("Please enter a valid email.")
+        return
+    } else {
+        setMessage("")
+    }
```
Comment (datderek) [inline pages/forgot-password.tsx:22]:
```suggestion
    if (email.length === 0) {
        setErrorMessage("Please enter a valid email.")
        return
    } else {
        setErrorMessage("")
    }
```

**Review by datderek — COMMENTED** [review]
Logic looks good so far. However, you may want to start migration to the App Router.

Also, could you edit the `api/create-account.js` and `/create-account.js`  to require an email during user sign-up?

**Comment by sammyoto** [comment]
I think redoing api/create-account.js could be a whole other issue we could put in the to do box? I would do it now but I'm going to be very busy until I leave on Tuesday.

**Review by datderek — APPROVED** [review]
Front-end looks good to me. 

I think we should switch to the **App Router** file structure after this task.

**Pull request merged**